File: src/com/sterndu/json/ByteValue.java
package com.sterndu.json;

import java.util.function.Function;

public class ByteValue implements NumberValue {

	private byte value = 0;

	public ByteValue(byte value) {
		this.value = value;
	}

	@Override
	public Number getValue() { return value; }

	public byte getValueByte() { return value; }

	public void setValue(byte value) { this.value = value; }

	@Override
	public String toJson() {
		return value + "";
	}

	@Override
	public String toJson(Function<Object, String> function) {
		return value + "";
	}

	@Override
	public String toString() {
		return toJson();
	}

}

File: src/com/sterndu/json/ByteValue.kt
@file:JvmName("ByteValue")
package com.sterndu.json


class ByteValue @JvmOverloads constructor(value: Byte = 0) : NumberValue {

	var valueByte: Byte = value
		private set

	override fun getValue(): Number {
		return valueByte
	}

	fun setValue(value: Byte) {
		valueByte = value
	}

	override fun toJson(): String {
		return valueByte.toString()
	}

	override fun toJson(function: (Any?) -> String): String {
		return valueByte.toString()
	}

	override fun toJsonValue(): JsonValue {
		return this
	}

	override fun toJsonValue(function: (Any?) -> String): JsonValue {
		return this
	}

	override fun toString(): String {
		return toJson()
	}
}
